Small cleanup

mrhota opened the pull request:
Hi again! just bumping some deps and cleaning some warnings before I jump back in. Hope you're doing well!

mrhota committed:
Bump libc-test and musl to latest and latest stable, respectively

mrhota committed:
Bump dependencies

mrhota committed:
Fix warnings

mrhota committed:
Remove cleanall from test script

mrhota committed:
Add rustfmt component using rustup

mrhota commented:
@anp apparently lots of the recent compiler refactors changed the way object files get named in the `.a` output file! so our `frankenstein.sh` script might not work the way it used to.

mrhota committed:
Don't move librusl.a, just copy

mrhota committed:
Add script to show disassembly of one function from an object

mrhota committed:
Use separate directory for test_rusl test runs

mrhota committed:
Clean before tests to prevent hidden failures

anp commented:
Hi! Looks like your malloc changes here are just syntactic, maybe upstream musl fixed some previously failing tests?

mrhota commented:
oh yeah, I just noticed changes to `mmap` which I think broke these malloc tests.

mrhota committed:
Fix broken malloc tests

A commit in Sept. to the mmap.c file fixed an OS
problem where mmap would sometimes return EPERM
instead of ENOMEM. This caused several malloc tests
to fail mysteriously. This ports the musl fix to
rusl.

mrhota commented:
@anp ok, I'll call this PR done now. What do you think?

mrhota commented on scripts/examine.sh:
```diff
@@ -0,0 +1,17 @@
+exec objdump -d "$@" |
+awk " /^[[:xdigit:]].*<$fn>/,/^\$/ { print \$0 }" |
+awk -F: -F' '  'NR==1 {  offset=strtonum("0x"$1); print $0; }
+                NR!=1 {  split($0,a,":"); rhs=a[2]; n=strtonum("0x"$1); $1=sprintf("%x", n-offset); printf "%4s:%s\n", $1,rhs }'
```
copied from stackoverflow!

mrhota committed:
Add reference to musl commit for EPERM fixup

anp reviewed (approved):
Awesome!

mrhota commented:
@anp I don't have merge permission, so you'll have to merge/squash these PRs

anp commented:
Oops!
